front/src/modules/settings/data-model/components/SettingsObjectFieldRelationForm.tsx

```diff
@@ -18,6 +18,7 @@ export type SettingsObjectFieldRelationFormValues = Partial<{
 }>;
 
 type SettingsObjectFieldRelationFormProps = {
+  disableFieldEdition?: boolean;
   disableRelationEdition?: boolean;
```

Comment:
disableRelationObjectEdition?

Reply:
This disables not only selecting the relation object, but also the relation type (Has many/Has one/Belongs to one).